nexus/types/src/deployment.rs

```diff
@@ -347,6 +344,23 @@ impl BpSledSubtableData for BlueprintOrCollectionZonesConfig {
     }
 }
 
+/// Iterate over the [`BlueprintZoneConfig`] instances that match the
+/// provided filter, along with the associated sled id.
+//
+// Ths was moved to a free function so that it could be used in the
```

Comment:
Thank you for the careful review! I think a strong type makes the most sense here, although I'm hoping it won't ripple much throughout the codebase.

Reply:
Alright, I changed my mind against a strong type. It will have too many ripple effects and I'm not sure it actually makes things clearer. I started to lean towards scoping this function to the Blueprint, but a method already exists with the same name. I still like this solution better than the free function though, so I'm going to rename it to `all_bp_zones`. If you think of a better name, I am happy to hear it.

Reply:
Changed in [8e2ee68](https://github.com/oxidecomputer/omicron/pull/6566/commits/8e2ee686172a7e68352fd2c72cb777179a1f5af5)

Reply:
Damnit, forgot to cleanup the comment. Doing that now.

Reply:
Comment fixed in [23185e8](https://github.com/oxidecomputer/omicron/pull/6566/commits/23185e831bb4593748ec19550bf7bf10cdd6f98d)

Reply:
> I still like this solution better than the free function though, so I'm going to rename it to all_bp_zones. If you think of a better name, I am happy to hear it.

Maybe since the scoped function doesn't actually accept a blueprint, it could be `zones_filtered` or something like that?

Reply:
Ah, I like that. I changed it to `filtered_zones` though.